trusted-services.mk

```diff
@@ -78,12 +78,18 @@ ffa-sp-all-clean: ffa-$1-sp-clean
 ffa-sp-all-realclean: ffa-$1-sp-realclean
 
 optee_os_sp_paths += $(TS_INSTALL_PREFIX)/$(SP_DIR)/bin/$3.$(SP_EXT)
+fip_sp_json_paths += $(TS_INSTALL_PREFIX)/$(SP_DIR)/json/$1.json
 endef
 
 ifeq ($(SP_PACKAGING_METHOD),embedded)
 # Add the list of SP paths to the optee_os config
 OPTEE_OS_COMMON_EXTRA_FLAGS += SP_PATHS="$(optee_os_sp_paths)"
 else ifeq ($(SP_PACKAGING_METHOD),fip)
+$(TS_INSTALL_PREFIX)/opteesp/json/sp_layout.json: ffa-sp-all
+	python $(TS_PATH)/tools/python/merge_json.py $@ $(fip_sp_json_paths)
+
+optee-os-common: $(TS_INSTALL_PREFIX)/opteesp/json/sp_layout.json
```

Comment:
This is a strange dependency. I can understand the `optee-os-common: ffa-sp-all` dependency above to avoid making changes in the main make file. However, this dependency is even broken, I assume that `ffa-sp-all` must complete before the `sp_layout.json` file can be generated.

Reply:
Isn't it the purpose of the recipe just above? (although the command does not contain the target which is ugly but I assume the Python script has a hard-coded output path?)

Reply:
You're right, I missed that.